### DISCUSSION TITLE
Using Grafana Global Variables

### DISCUSSION THREAD

**Discussion by yanhan92** [grafadruid/druid-grafana#102] [Q&A]
Hi,

I am trying to test some simple queries using this plugin and running into some difficulty in accessing global variables in the sql.

For example, when i try:
`SELECT count(code) FROM "dw_diff-test" where __time >= TIMESTAMP '2021-08-10'
`
I get a result as expected:
![image](https://user-images.githubusercontent.com/13114045/130769768-0fd66c03-2a95-4322-a623-713d4858e0cf.png)

However, when i replace the timestamp with a global variable like:
`SELECT count(code) as C FROM "dw_diff-test" where __time >= TIMESTAMP '${__from:date}'`
or 
`SELECT count(code) as C FROM "dw_diff-test" where __time >= TIMESTAMP ${__from:date}`
or
`SELECT count(code) as C FROM "dw_diff-test" where __time >= ${__from:date}`

I get an error:
![image](https://user-images.githubusercontent.com/13114045/130770261-1bc25b08-c112-42d6-a431-ee3ca6bbec19.png)

In the query inspector, i see that the variable has not been replaced in the query (not sure if it should be):
![image](https://user-images.githubusercontent.com/13114045/130770453-f996a46e-6f29-4fe4-aea4-76c9c0f200bf.png)

May i know what am i doing wrong here?

**Comment by jermudgeon** [comment]
I'm having similar difficulties. I'm going to resort to inspecting the query in druid logs as I don't see a way to get a dump of the query with substitutions.

**Comment by jbguerraz** [comment]
Hello mates,

You need to wrap the variable with simple quotes: `__time >= '${__from:date}' AND __time <= '${__to:date}'`
Also you can get a dump of the query by looking at `data.queries[0].builder.query`

![image](https://user-images.githubusercontent.com/861556/133237237-009a6184-4be8-4d49-8d25-3b4d20d106e0.png)

Hope it helps :)

Note: this relates to https://github.com/grafadruid/druid-grafana/issues/98 (any contribution is welcome!)

**Reply by yanhan92** [reply]
Hi @jbguerraz,

Thank you for your reply. I tried what you suggested but i see that the time mapped into the SQL command is different from what it actually is.

![image](https://user-images.githubusercontent.com/13114045/135066006-9714873e-4de9-48d3-a814-85728edf2a4b.png)

Seems like a strange issue. Is it just me doing something silly?

p.s. i just refreshed the query inspector again and the sql still looks the same

Thank you.

Yan Han

**Reply by jbguerraz** [reply]
Hello @yanhan92  :wave: 
Interesting issue!! I've asked the Grafana mates since I don't understand this issue: https://grafana.slack.com/archives/C3HJV5PNE/p1632834011084200 
Basically, the `templateSrv` service provided by Grafana won't replace time range variables properly when in explore (but it's ok in dashboards).
I'll keep you posted!

**Reply by yanhan92** [reply]
Hey,

Thank you for your help. Ill try it with dashboards.
